### ISSUE TITLE
Override checkbox color

### ISSUE THREAD

**Issue by stianjensen** [react-toolbox/react-toolbox#791]
Theming components in react-toolbox is easy using css and css modules, but sometimes a one-off styling is required. Say, you have a list of checkboxes and want to style each of them in separate colors.

Currently, you would need to create a class for each color you want to set and then apply that as a className for the root Checkbox component. But since the checkbox itself is nested within and can't be controlled directly, you need to specify it with nested selectors:

```
.colorRed .check {
    background-color: red;
    border-color: red;
}

<Checkbox className={theme.colorRed} theme={theme} />
```

Even though inline styling is bad for general situations, in these cases inline colors are useful to override specific elements and avoid creating a bunch of specific classes (especially when colors are generated in js based on data).
It is possible to set the style attribute of the Checkbox element, but all those styles are currently applied to the internal input element, which is not what is seen as the colored checkbox.

Would it be possible to for instance add a color attribute to the Checkbox component to allow for just this situation? The value of the color attribute could then be applied as inline styling on the internal Check component as background and border color.

I fully agree with this project's decision to not use inline styling, and that is the exact reason why I moved from material-ui to react-toolbox, but I still believe that inline styles can be useful in cases like what I've described. If you agree that it could be useful or have suggestions for alternatives, I'd be happy to help out with a pull request.

**Comment by javivelasco** [comment]
Yeah I agree sometimes inline styles are useful but, as I see it, the main issue is that you have no way to access components that are buried under their parents. In my opinion passing down color for this component is like a very specific solution. In cases like this where you don't want to create a module just to hold a very small theme you can also take advantage of the `[data-react-toolbox]` attributes or revealing internal structure but if you are generating colors in the Javascript source and those colors are not a limited pre-defined set, inline is the way to go.

Now, for Checkboxes we are spreading props in the `input` element because most properties you pass to the component are likely to be placed there. Style is a special case. What do you think if we move the style attribute to be used in the checkbox element? I did not checked the source yet, and don't know if it solves the issue but I wanted to drop a comment before checking deeply

**Comment by stianjensen** [comment]
For my specific use case, moving the style prop to the checkbox element sounds like a good fix. And I think that would make sense, seeing as that is the element which is visible.

I totally agree that creating specific props like color for all sorts of special cases would be a bad solution.

**Comment by stianjensen** [comment]
I tried creating a pull request for this. Let me know if this is what you had in mind or any changes you'd like to make 😉

**Comment by matthewberkompas** [comment]
How soon will it be before this is implemented @javivelasco? I have a situation where I need to set the color of the checkboxes dynamically and this would really help. 
Great library by the way! It's well documented and super easy to use.

**Comment by javivelasco** [comment]
Done! Thanks for your words @matthewberkompas and sorry for the delay!

**Comment by matthewberkompas** [comment]
This doesn't seem to have fixed the problem. For me inline `styles` are still being set on the `input` element instead of the `div`. Is it possible I have the wrong version number? I'm currently on version 1.2.1 which matches the one [here](https://github.com/react-toolbox/react-toolbox). Any ideas?

**Comment by javivelasco** [comment]
It is not release yet. You have to either wait for a next patched release or to use directly the dev branch from the repository!

**Comment by matthewberkompas** [comment]
Got it. Not to bug you or anything, but how soon until this change is released? Thanks a lot!

**Comment by javivelasco** [comment]
I'm releasing today :)

**Comment by matthewberkompas** [comment]
Is this working for you now @stianjensen? After updating my react-toolbox version to 1.2.2, I'm still seeing the `style` prop being applied to the `input` element and not the `.check` div.

**Comment by stianjensen** [comment]
It works for me after updating, yes. Are you sure you are using the latest version – try deleting your `node_modules` for instance.

Otherwise, post some code here and I can try and figure out what's wrong.

**Comment by matthewberkompas** [comment]
My version is 1.2.2 which was released 2 days ago.
```
{
            this.data.map((key, index) => {
              return (
                <Checkbox
                  key={`legend-${index}`}
                  label={key.toUpperCase()}
                  checked={this.state.checked}
                  onChange={() => this.handleToggle(key)}
                  theme={theme}
                  style={{backgroundColor: `${timePeriodColors[index]}`}}
                />
              );
            })
          }
```
I'm trying to set the background color of each checkbox to the color matching it's respective index in my `timePeriodColors` array. When I inspect the checkbox in Chrome however, I see that this style is getting set on the `input` element instead of the div with `data-react-toolbox="check"`.

**Comment by stianjensen** [comment]
@matthewberkompas Looking at the code here, which is the code that ships with 1.2.2 everything is looking correct. Style is passed down to the Check element.

https://github.com/react-toolbox/react-toolbox/blob/1.2.2/lib/checkbox/Checkbox.js#L104

I also confirmed this by testing right now with installing react-toolbox 1.2.2 in a clean new project.

Can you confirm your installed version of react-toolbox by going to `node_modules/react-toolbox/package.json` and read the version there?

**Comment by matthewberkompas** [comment]
Yeah, your code looks right. After checking `node_modules/react-toolbox/package.json`, I found that my version still appears to be 1.1.2. Running, `npm update react-toolbox` after I changed the version number in my main `package.json` didn't make any difference. Is there another command that I should run?

**Comment by stianjensen** [comment]
After updating `package.json` you can simply run `npm install` to update all packages to their specified version.

**Comment by matthewberkompas** [comment]
I actually just discovered what the problem was. Since I'm working on a project that is primarily written in ruby on rails, all of the react code is in a sub-directory. It turns out that I was in the wrong directory when I ran `npm update react-toolbox` after updating the version number so it didn't actually do anything 😛 . Now it's working great:
<img width="395" alt="screen shot 2016-10-10 at 12 54 43 pm" src="https://cloud.githubusercontent.com/assets/7084294/19249150/c489511a-8ee8-11e6-9039-83168a9d598d.png">
Thanks a lot for your time @stianjensen. I really appreciate it.
